test/http_test.js

```diff
@@ -14,161 +12,192 @@ chai.config.includeStack = true;
     });
 
-    it("should resolve with null JSON if Content-Length is '0'", () => {
-      sandbox.stub(root, "fetch").returns(
-        fakeServerResponse(200, "", {"Content-Length": "0"}));
+    describe("Backoff header handling", () => {
+      var http;
 
-      return request("/")
-        .then(res => res.json)
-        .should.eventually.become(null);
-    });
-  });
+      beforeEach(() => {
+        // Make utils.getUnixTime to always return 1000, for easier computation
+        sandbox.stub(Date.prototype, "getTime").returns(1000 * 1000);
+        http = new HTTP();
+      });
 
-  describe("Malformed JSON response", () => {
-    it("should reject with an appropriate message", () => {
-      sandbox.stub(root, "fetch").returns(Promise.resolve({
-        status: 200,
-        headers: {
-          get(name) {
-            if (name !== "Alert")
-              return "fake";
-          }
-        },
-        json() {
-          return JSON.parse("malformed json");
-        }
-      }));
-
-      return request("/")
-        .should.be.rejectedWith(Error, /HTTP 200; SyntaxError: Unexpected token/);
-    });
-  });
+      it("should emit a backoff event on set Backoff header", () => {
+        sandbox.stub(root, "fetch").returns(
+          fakeServerResponse(200, {}, {Backoff: "1000"}));
 
-  describe("Business error responses", () => {
-    it("should reject on status code > 400", () => {
-      sandbox.stub(root, "fetch").returns(
-        fakeServerResponse(400, {
-          code: 400,
-          details: [
-            {
-              description: "data is missing",
-              location: "body",
-              name: "data"
-            }
-          ],
-          errno: 107,
-          error: "Invalid parameters",
-          message: "data is missing"
-      }));
-
-      return request("/")
-        .should.be.rejectedWith(Error, /HTTP 400; Invalid request parameter: data is missing/);
-    });
-  });
-
-  describe("Deprecation header", () => {
-    beforeEach(() => {
-      sandbox.stub(console, "warn");
-    });
-
-    it("should handle deprecation header", () => {
-      const eolObject = {
-        code:    "soft-eol",
-        url:     "http://eos-url",
-        message: "This service will soon be decommissioned",
-      };
-      sandbox.stub(root, "fetch").returns(
-        fakeServerResponse(200, {}, {Alert: JSON.stringify(eolObject)}));
-
-      return request("/")
-        .then(_ => {
-          sinon.assert.calledOnce(console.warn);
-          sinon.assert.calledWithExactly(
-            console.warn, eolObject.message, eolObject.url);
-        });
-    });
+        return http.on("backoff", value => {
+          expect(value).eql(2000);
+        }).request("/");
+      });
 
-    it("should handle deprecation header parse error", () => {
-      sandbox.stub(root, "fetch").returns(
-        fakeServerResponse(200, {}, {Alert: "dafuq"}));
+      it("should emit a backoff event on missing Backoff header", () => {
```

Comment:
What does it means it should emit a backoff event when no Backoff header? To stop the backoff. Is this to release backoff when no more backoff are sent?